apple/testing/default_runner/ios_xctestrun_runner.template.sh

```diff
@@ -128,17 +144,26 @@ if [[ -n "$test_host_path" ]]; then
     xcrun_target_app_path="$xctestrun_test_host_path"
     # If ui testing is enabled we need to copy out the XCTRunner app, update its info.plist accordingly and finally
     # copy over the needed frameworks to enable ui testing
-    readonly runner_app_name="XCTRunner"
+    readonly runner_app_name="$test_bundle_name-Runner"
     readonly runner_app="$runner_app_name.app"
     readonly runner_app_destination="$test_tmp_dir/$runner_app"
-    libraries_path="$(xcode-select -p)/Platforms/iPhoneSimulator.platform/Developer/Library"
+    developer_path="$(xcode-select -p)/Platforms/$test_execution_platform/Developer"
+    libraries_path="$developer_path/Library"
     cp -R "$libraries_path/Xcode/Agents/XCTRunner.app" "$runner_app_destination"
     chmod -R 777 "$runner_app_destination"
     xctestrun_test_host_path="__TESTROOT__/$runner_app"
+    xcrun_test_host_bundle_identifier="com.apple.test.$runner_app_name"
+    plugins_path="$test_tmp_dir/$runner_app/PlugIns"
+    mkdir -p "$plugins_path"
+    cp -R "$test_tmp_dir/$test_bundle_name.xctest" "$plugins_path"
+    mkdir "$plugins_path/$test_bundle_name.xctest/Frameworks"
+    # We need this dylib for 14.x OSes
+    cp "$(xcode-select -p)/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot/usr/lib/swift/libswift_Concurrency.dylib" "$plugins_path/$test_bundle_name.xctest/Frameworks/libswift_Concurrency.dylib"
```

Comment:
Suggestion on comment.

Happen to know if this breaks x86_64 simulator UI test execution?

```suggestion
    # We need this dylib for 14.x OSes. This intentionally doesn't use `test_execution_platform`
    # since this file isn't present in the `iPhoneSimulator.platform`.
    cp "$(xcode-select -p)/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot/usr/lib/swift/libswift_Concurrency.dylib" "$plugins_path/$test_bundle_name.xctest/Frameworks/libswift_Concurrency.dylib"
```